cas-server-support-basic/src/main/java/org/jasig/cas/web/flow/BasicAuthenticationAction.java

```diff
@@ -0,0 +1,92 @@
+package org.jasig.cas.web.flow;
+
+import org.apache.commons.codec.binary.Base64;
+import org.apache.http.HttpStatus;
+import org.jasig.cas.CentralAuthenticationService;
+import org.jasig.cas.authentication.UsernamePasswordCredential;
+import org.jasig.cas.ticket.TicketGrantingTicket;
+import org.jasig.cas.web.support.WebUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.stereotype.Component;
+import org.springframework.webflow.action.AbstractAction;
+import org.springframework.webflow.execution.Event;
+import org.springframework.webflow.execution.RequestContext;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.io.UnsupportedEncodingException;
+import java.util.StringTokenizer;
+
+/**
+ * This is {@link BasicAuthenticationAction}.
+ *
+ * @author Misagh Moayyed
+ * @since 4.2.0
+ */
+@Component("basicAuthenticationAction")
+public class BasicAuthenticationAction extends AbstractAction {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(BasicAuthenticationAction.class);
+
+    private static final String AUTHZ_HEADER = "Authorization";
+
+    @Autowired
+    @Qualifier("centralAuthenticationService")
+    private CentralAuthenticationService centralAuthenticationService;
+
+    @Override
+    protected Event doExecute(final RequestContext requestContext) throws Exception {
+
+        final HttpServletRequest request = WebUtils.getHttpServletRequest(requestContext);
+        final HttpServletResponse response = WebUtils.getHttpServletResponse(requestContext);
+
+        final String authHeader = request.getHeader(AUTHZ_HEADER);
```

Comment:
You could certainly use this pac4j class: https://github.com/pac4j/pac4j/blob/master/pac4j-http/src/main/java/org/pac4j/http/credentials/extractor/BasicAuthExtractor.java if you are not reluctant to add this dependency, to extract credentials.
I realized this needs to be improved BTW: https://github.com/pac4j/pac4j/issues/395